bundles/binding/org.openhab.binding.onkyo/src/main/java/org/openhab/binding/onkyo/internal/eiscp/EiscpSerial.java

```diff
@@ -0,0 +1,261 @@
+			}
+	
+			// RXTX serial port library causes high CPU load
+			// Start event listener, which will just sleep and slow down event loop
+			serialPort.addEventListener(this);
+			serialPort.notifyOnDataAvailable(true);
+			return true;
+		} catch (Exception e) {
+	 	    logger.error("serial port connect error", e);
+		    e.printStackTrace();
+		    return false;
+		}
+
+	}
+
+	/**
+	 * {@inheritDoc}
+	 */
+	public boolean closeSocket() {
+	    try{
+		if (out != null) {
+			logger.debug("Close serial out stream");
+			IOUtils.closeQuietly(out);
+		}
+		if (in != null) {
+			logger.debug("Close serial in stream");
+			IOUtils.closeQuietly(in);
+		}
+		if (serialPort != null) {
+			logger.debug("Close serial port");
+			serialPort.close();
+		}
+		
+		serialPort.removeEventListener();
+		
+		serialPort = null;
+		out = null;
+		in = null;
+
+		logger.debug("Closed");
+		return true;
+		}catch(Exception e){
+			logger.error("Unable to close ", e);
+			return false;
+		}
+	}
+
+        private StringBuilder getEiscpMessage(String eiscpCmd) {
+		logger.debug("Requested Command is "+eiscpCmd);
+	        StringBuilder sb = new StringBuilder();
+		sb.append("!");
+		sb.append("1");
+		sb.append(eiscpCmd);
+		sb.append((char)0x0D);
+		logger.debug("Serial command is "+sb);
+		return sb;
+	}
+
+
+	/**
+	 * {@inheritDoc}
+	 */
+	public void sendCommand(String eiscpCmd) {
+		String data = getEiscpMessage(eiscpCmd).toString();
+		if (in == null || out == null) {
+			connectSocket();
+		}
+		
+		try {
+			// flush input stream
+			if (in.markSupported()) {
+				in.reset();
+			} else {
+
+				while (in.available() > 0) {
+
+					int availableBytes = in.available();
+
+					if (availableBytes > 0) {
+
+						byte[] tmpData = new byte[availableBytes];
+						in.read(tmpData, 0, availableBytes);
+					}
+				}
+			}
+
+	
+			String response = sendMmsg(data, timeout);
+			if (response!=null)
+				notifier.notifyMessage(new OnkyoStatusUpdateEvent(this), response.getBytes(), 8);
+			
+		} catch (IOException e) {
+			
+			logger.debug("IO error occured...reconnect and resend ones");
+			closeSocket();
+			connectSocket();
+			
+			try {
+				String response = sendMmsg(data, timeout);
+				if (response!=null)
+					notifier.notifyMessage(new OnkyoStatusUpdateEvent(this), response.getBytes(), 8);
+
+			} catch (IOException e1) {
+				logger.error("Send failed.."+e1);
+			}catch(EiscpException ee){
+				logger.error("Unable to notify "+ee);
+			}
+			
+		} catch (Exception e) {
+				logger.error("Send failed.."+e);
+		}
+	}
+
+	@Override
+	public void serialEvent(SerialPortEvent portEvent) {
+		try {
+			logger.debug("Received serial port event "+portEvent.toString()+":"+portEvent.getEventType());
+			logger.trace("RXTX library CPU load workaround, sleep forever");
+			Thread.sleep(Long.MAX_VALUE);
+		} catch (  InterruptedException  e) {
+			logger.warn("Unable to notify", e);
+		}
+	}
+	
+	private String sendMmsg(String data, int timeout) throws IOException {
```

Comment:
Is `sendMmsg` the intended spelling, or should it be `sendMsg` or `sendMessage`?